Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py

```diff
@@ -1014,13 +1014,32 @@ def parse_custom_fields(extensions):
                 break
         return custom_fields, score
 
+    @staticmethod
+    def get_single_pattern_value(pattern: str) -> str:
+        """
+        Parses a pattern with a single single key & value pair.
+        For Example:
+
+        >>> STIX2XSOARParser.get_single_pattern_value("[domain-name:value = 'www.example.com']")
+        'www.example.com'
+
+        Args:
+            pattern: the pattern to extract the value from.
+
+        Returns:
+            str. the value in the pattern.
+
+
+        """
+        return pattern.rpartition(' = ')[2].strip(" ']")
```

Comment:
I understand but the current implementation will retrieve the latest name in case we are getting a complex pattern so is this the desired behavior or should we retrieve the first one?
> current implementation will retrieve the latest name in case we are getting a complex pattern so is this the desired behavior or should we retrieve the first one?

Reply:
The latest or the first? First is better.

Reply:
@jlevypaloalto So let's change the rpartition to split and take the first one as @Ni-Knight suggested.

Reply:
We'll use the `stix2pattern` library to parse.